packages/vuetify/src/labs/VVideo/VVideoControls.tsx

```diff
@@ -0,0 +1,298 @@
+  // disabled: Boolean,
+  playing: Boolean,
+  hidePlay: Boolean,
+  hideVolume: Boolean,
+  hideFullscreen: Boolean,
+  fullscreen: Boolean,
+  floating: Boolean,
+  splitTime: Boolean,
+  pills: Boolean,
+  detached: Boolean,
+  progress: {
+    type: Number,
+    default: 0,
+  },
+  duration: {
+    type: Number,
+    default: 0,
+  },
+  volume: [Number, String],
+  variant: {
+    type: String as PropType<VVideoControlsVariant>,
+    default: 'default',
+    validator: (v: any) => allowedVariants.includes(v),
+  },
+  volumeProps: Object as PropType<Pick<VVideoVolume['$props'], 'direction' | 'inline' | 'sliderProps' | 'menuProps'>>,
+
+  ...makeDensityProps(),
+  ...makeElevationProps(),
+  ...makeThemeProps(),
+}, 'VVideoControls')
+
+export const VVideoControls = genericComponent<VVideoControlsSlots>()({
+  name: 'VVideoControls',
+
+  props: makeVVideoControlsProps(),
+
+  emits: {
+    'update:playing': (val: boolean) => true,
+    'update:progress': (val: number) => true,
+    'update:volume': (val: number) => true,
+    skip: (val: number) => true,
+    'click:fullscreen': () => true,
+  },
+
+  setup (props, { emit, slots }) {
+    const { themeClasses } = provideTheme(props)
+    const { densityClasses } = useDensity(props)
+    const { elevationClasses } = useElevation(props)
+
+    const { backgroundColorClasses, backgroundColorStyles } = useBackgroundColor(() => {
+      const fallbackBackground = props.detached ? 'surface' : undefined
+      return props.backgroundColor ?? fallbackBackground
+    })
+
+    const playing = useProxiedModel(props, 'playing')
+    const progress = useProxiedModel(props, 'progress')
+    const volume = useProxiedModel(props, 'volume', 0, (v?: number | string) => Number(v ?? 0))
+    const lastVolume = shallowRef(0)
+
+    const currentTime = computed(() => {
+      const secondsElapsed = Math.round(props.progress / 100 * props.duration)
+      return {
+        elapsed: formatTime(secondsElapsed),
+        remaining: formatTime(props.duration - secondsElapsed),
+        total: formatTime(props.duration),
+      }
+    })
+
+    function play () {
+      playing.value = true
+    }
+
+    function pause () {
+      playing.value = false
+    }
+
+    function skipTo (v: number) {
+      progress.value = v
+    }
+
+    function toggleMuted () {
+      if (volume.value) {
+        lastVolume.value = volume.value
+        volume.value = 0
+      } else {
+        volume.value = lastVolume.value
+      }
+    }
+
+    function toggleFullscreen () {
+      emit('click:fullscreen')
+    }
+
+    useRender(() => {
+      const sizes = props.pills
+        ? [42, 36, 30]
+        : [32, 28, 24]
+
+      const innerDefaults = {
+        VIconBtn: {
+          size: props.density === 'compact' ? sizes[2]
+          : props.density === 'comfortable' ? sizes[1]
+          : sizes[0],
+          iconSize: props.density === 'compact' ? 20
+          : props.density === 'comfortable' ? 24
+          : 26,
```

Comment:
Not the job of this P.R., but this is something we will need to look into how to do in a clearer way. There are a lot of non configurable magic numbers here.

Reply:
Yes, I opted for a simpler code to ship it faster. Once it stabilizes a bit, we might figure something out.